lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -2,42 +2,73 @@ module ActiveModel
   class Serializer
     module Adapter
       class Attributes < Base
+        attr_accessor :_nested_associations
+
+        def initialize(serializer, options = {})
+          options ||= {}
+
+          self._nested_associations = ActiveModel::Serializer::Utils
+            .include_args_to_hash(options[:include])
+
+          super(serializer, options)
```

Comment:
`render json: object, include: [whatever]` passes `Adapter.new(options = { include: [whatever])` and then the Rails Renderer calls `adapter.to_json` more or less